DDD-10.md

```diff
@@ -0,0 +1,112 @@
+# DBZ-8968: Add support for pluggable ArrayDeque
+
+## Motivation
+
+Debezium (DBZ) connectors use an in-memory queue (ArrayDeque) within the ChangeEventQueue class to buffer changes between the database reader and the Kafka Connect task. When the Connect task (Eg: IcebergConsumer) is slow to process events, the in-memory queue can become full, causing the database reader to slow down due to backpressure.
+
+For example, The Postgres Connector is single-threaded and memory-bound, which limits throughput. 
+In snapshot phase, DB reads can be parallelized using the `snapshot.fetch.threads` config. However, the benefits are limited without a fast and efficient writer because of limited internal queue capacity.
+
+This behavior introduces limitations, including:
+1. Memory pressure and frequent garbage collection
+2. OOM risks under high load (max.queue.size.in.bytes introduces additional overhead)
```

Comment:
Do we really have many reports of OOM issues practically?

Reply:
It has the potential, more so in some connectors than others. And as we start to consider introducing a transaction buffer for Informix and PostgreSQL, that potential will increase.